backend/src/openarchiefbeheer/destruction/api/viewsets.py

```diff
@@ -221,18 +224,50 @@ class DestructionListViewSet(
     viewsets.GenericViewSet,
 ):
     serializer_class = DestructionListWriteSerializer
-    queryset = (
-        DestructionList.objects.all()
-        .select_related("author", "assignee")
-        .prefetch_related(
-            Prefetch(
-                "assignees",
-                queryset=DestructionListAssignee.objects.prefetch_related(
-                    "user__user_permissions"
-                ).order_by("pk"),
-            )
-        )
+    user_prefetch = User.objects.prefetch_related(
+        Prefetch(
+            "user_permissions",
+            queryset=Permission.objects.select_related("content_type").order_by(
+                "codename"
+            ),
+        ),
+        Prefetch(
+            "groups",
+            queryset=Group.objects.prefetch_related(
+                Prefetch(
+                    "permissions",
+                    queryset=Permission.objects.select_related("content_type").order_by(
+                        "codename"
+                    ),
+                )
+            ),
+        ),
+    ).annotate(
+        user_permission_codenames=ArrayAgg("user_permissions__codename", distinct=True),
+        group_permission_codenames=ArrayAgg(
+            "groups__permissions__codename", distinct=True
+        ),
     )
```

Comment:
Maybe this could live somewhere in the `UserManager`? Then you could call `User.objects.all_with_permissions_data()` (or something like that, but a better name :sweat_smile: )?

Reply:
I tried using a custom QuerySet, but ran into issues because of `UserManager`, I can have another attempt and if it's doesnt work I'll try the Manager 👍